examples/apps/ai-collab/src/components/TaskCard.tsx

```diff
@@ -78,26 +77,37 @@ export function TaskCard(props: {
 
 	const fieldDifferences: {
 		isNewCreation: boolean;
-		changes: Record<string, DifferenceChange>;
-		moved?: DifferenceMove;
+		changes: Record<string, ModifyDiff>;
+		moved?: {
+			uiDiff: MoveDiff;
+			originalIndex: number;
+		};
 	} = {
 		isNewCreation: false,
-		changes: {} satisfies Record<string, DifferenceChange>,
+		changes: {} satisfies Record<string, ModifyDiff>,
 	};
 
-	for (const diff of branchDifferences ?? []) {
-		if (diff.type === "CHANGE") {
-			const path = diff.path[diff.path.length - 1];
-			if (path === undefined) {
-				throw new Error(`List of paths in CHANGE diff is empty`);
-			}
-			fieldDifferences.changes[path] = diff;
-		}
-		if (diff.type === "CREATE") {
+	for (const uiDiff of uiDiffs ?? []) {
+		if (uiDiff.type === "insert") {
 			fieldDifferences.isNewCreation = true;
 		}
-		if (diff.type === "MOVE") {
-			fieldDifferences.moved = diff;
+		if (uiDiff.type === "modify") {
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+			const targetField = uiDiff.nodePath[0]?.parentField as string;
+			if (targetField === undefined) {
+				console.error("Recieved modify ui diff but could not identify target field");
+				continue;
+			}
+
+			fieldDifferences.changes[targetField] = uiDiff;
+		}
+
+		if (uiDiff.type === "move" && uiDiff.subType === "move-single") {
```

Comment:
Do we need handling of a `move-range` in addition to a `move-single`?

Reply:
Only create and update operations are showing on Ui now